6_nodejs/week2/app.js

```diff
@@ -0,0 +1,94 @@
+const express = require("express");
+const fs = require("fs");
+const app = express();
+const port = process.env.PORT || 3000;
+
+app.use(express.json());
+
+app.get("/search", (req, res) => {
+  const query = req.query.q;
+
+  try {
+    const data = fs.readFileSync("./documents.json", "utf8");
+    const documents = JSON.parse(data);
+
+    if (!query) {
+      res.json(documents);
+    } else {
+      const filteredDocuments = documents.filter((doc) => {
```

Comment:
This could be optimized using **some**. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some